lib/src/drawing_and_graphics/shadow.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:ui';
+
+import 'package:flutter/widgets.dart';
+
+/// A widget that adds a shadow effect to its child widget.
+class Shadow extends StatelessWidget {
+  /// Creates a Shadow widget with the given properties.
+  ///
+  /// Use the [radius] to set the amount of blur, where `0.0` is sharp and a higher
+  /// value is more fuzzy. You can also set the shadow [color] and ([x], [y]) offset.
+  const Shadow({
+    super.key,
+    // In SwiftUI the default is `Color(.sRGBLinear, white: 0, opacity: 0.33)`
+    this.color = const Color(0x54000000),
+    required this.radius,
+    this.x = 0,
+    this.y = 0,
+    required this.child,
+  });
+
+  /// The color of the shadow.
+  final Color color;
+
+  /// The blur radius of the shadow.
+  ///
+  /// `0.0` is sharp while higher values are more fuzzy.
+  final double radius;
+
+  /// The horizontal offset of the shadow.
+  final double x;
+
+  /// The vertical offset of the shadow.
+  final double y;
+
+  /// The widget to which the shadow effect will be applied.
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        Transform.translate(
```

Comment:
Can you check what Swift UI does with semi-translucent shapes that have a shadow applied?

Reply:
When applying a shadow to a semi-translucent shape in SwiftUI, the shadow itself appears to retain its opacity:

<img width="176" alt="Screenshot 2024-07-02 at 16 48 50" src="https://github.com/Flutter-Bounty-Hunters/swift_ui/assets/10300220/f60f8e89-303a-412e-9e55-7eeff2123030">

```swift
Ellipse()
    .fill(Color.yellow)
    .shadow(color: .gray, radius: 10, x: 10, y: 10)
    .frame(width: 200, height: 100)

Ellipse()
    .fill(Color.yellow.opacity(0.3))
    .shadow(color: .gray, radius: 10, x: 10, y: 10)
    .frame(width: 200, height: 100)
```

This is not the case with the current implementation in Flutter. The shadow is also lightened based on the child's opacity:

<img width="247" alt="Screenshot 2024-07-02 at 16 53 27" src="https://github.com/Flutter-Bounty-Hunters/swift_ui/assets/10300220/0ba43ce9-fc2a-4b42-a274-e31353072a84">

```dart
Frame(
  width: 200,
  height: 100,
  Shadow(
    color: color,
    radius: radius,
    x: x,
    y: y,
    Ellipse(
      fillColor: Colors.yellow,
    ),
  ),
),

Frame(
  width: 200,
  height: 100,
  Shadow(
    color: color,
    radius: radius,
    x: x,
    y: y,
    Ellipse(
      fillColor: Colors.yellow.withOpacity(0.3),
    ),
  ),
),
```

Should we match the SwiftUI version? In my opinion, the Flutter version looks better, but if we want a pixel perfect copy, the answer is probably yes. If so, I'll need to think about how to do that.

Reply:
If possible, let's do what SwiftUI does. The Flutter example that you included looks like what I'd expect if I applied an opacity around the whole widget tree, including the shadow. But if I only applied opacity to the shape's color, I wouldn't expect that to impact the shadow.

Reply:
I updated the widget implementation so that the widget opacity and shadow opacity are now independent of each other. The flutter version now works like the SwiftUI version for partial transparency in the widget but full opacity for the inner shadow color:

<img width="265" alt="Screenshot 2024-07-25 at 04 48 13" src="https://github.com/user-attachments/assets/dc6e19ed-579a-4b9f-8743-7f0f0de5d7e9">

The implementation had the side effect of removing antialiasing from the edges before the blurring step. So if the blur radius was 0.0, the jagged edges were noticeable. So in the case when radius is 0.0, I now retain the original widget opacity (including its antialiasing) for the shadow. That means there still is a rare edge case where someone uses a semitransparent widget with a 0-radius shadow. In that case the shadow would also be semitransparent. This would be quite rare and a slightly more transparent shadow wouldn't look bad (in my opinion), so I recommend we don't worry about it for now.